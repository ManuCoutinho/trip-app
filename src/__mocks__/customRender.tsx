import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
