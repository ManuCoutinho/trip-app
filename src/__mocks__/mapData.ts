import { MapProps } from 'components/Map/types'

export default {
  places: [
    {
      id: 'first-place',
      name: 'Rio de Janeiro',
      slug: 'rio-de-janeiro',
      location: {
        latitude: -22.925453,
        longitude: -43.277763
      }
    },
    {
      id: 'second-place',
      name: 'Lagos',
      slug: 'lagos',
      location: {
        latitude: 6.460649,
        longitude: 3.333309
      }
    }
  ]
} as MapProps
