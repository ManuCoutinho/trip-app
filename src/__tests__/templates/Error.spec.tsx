import { render } from '__mocks__/customRender'
import { fireEvent, renderHook } from '@testing-library/react'
import mockRouter from 'next-router-mock'
import { act } from 'react-dom/test-utils'
import ErrorTemplate from 'templates/Error'
import { useRouter } from 'next/router'

jest.mock('next/router', () => require('next-router-mock'))
const templatePropsMock = {
  statusCode: 500
}
describe('<ErrorTemplate />', () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl('/about')
  })
  it('should render the Error template correctly', () => {
    const { getByRole, getByText } = render(
      <ErrorTemplate {...templatePropsMock} />
    )
    expect(getByRole('img', { name: /error/i })).toBeInTheDocument()
    expect(getByText('Thanks for the patience!')).toBeInTheDocument()
    expect(
      getByText(`Error ${templatePropsMock.statusCode}`)
    ).toBeInTheDocument()
  })
  it('should navigate to home page when click in button', () => {
    const { result } = renderHook(() => useRouter())
    const { getByRole } = render(<ErrorTemplate {...templatePropsMock} />)
    const button = getByRole('button', { name: /home/i })
    act(() => {
      fireEvent.click(button)
    })
    expect(result.current.pathname).toBe('/')
  })
  it('should omit status code in the document', () => {
    const { queryByText } = render(<ErrorTemplate statusCode={null} />)

    expect(
      queryByText(`Error ${templatePropsMock.statusCode}`)
    ).not.toBeInTheDocument()
  })
  it('should match to snapshot', () => {
    const { container } = render(<ErrorTemplate {...templatePropsMock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
