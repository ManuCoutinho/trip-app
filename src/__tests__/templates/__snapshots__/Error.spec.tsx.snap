// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorTemplate /> should match to snapshot 1`] = `
<section
  class="sc-bcXHqe bViWhL"
>
  <div
    class="sc-dkrFOg ihBGEd"
  >
    <img
      alt="error image"
      data-nimg="1"
      decoding="async"
      height="500"
      loading="lazy"
      src="/assets/error.svg"
      srcset="/assets/error.svg 1x, /assets/error.svg 2x"
      style="color: transparent;"
      width="500"
    />
  </div>
  <div
    class="sc-gswNZR Fbifm"
  >
    <p>
      Error 500
    </p>
    <h3
      class="sc-jSUZER ikUjKO"
    >
      Oops! Something unexpected has occurred!
    </h3>
    <p
      class="sc-hLBbgP fTgiIa"
    >
       We're sorry for the inconvenience, but it looks like the page you're
            trying to access is down for maintenance.
    </p>
    <p
      class="sc-hLBbgP fTgiIa"
    >
      Our team is already working to normalize access.
    </p>
    <p
      class="sc-hLBbgP fTgiIa"
    >
      Thanks for the patience!
    </p>
    <button
      class="sc-eDvSVe jeDjtr"
      type="button"
    >
      Home
    </button>
  </div>
</section>
`;
